colorpicker/src/main/java/com/thebluealliance/colorpicker/ColorPreference.java

```diff
@@ -0,0 +1,185 @@
+package com.thebluealliance.colorpicker;
+
+import com.thebluealliance.colorpicker.internal.ColorCircleDrawable;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.res.TypedArray;
+import android.graphics.Color;
+import android.os.Build;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.preference.DialogPreference;
+import android.support.annotation.ColorInt;
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.ViewGroup;
+
+public class ColorPreference extends DialogPreference {
+
+    private static final @ColorInt int DEFAULT_VALUE = Color.BLACK;
+
+    private @ColorInt int[] mColors;
+    private @ColorInt int mCurrentValue;
+    private ColorPalette mColorPalette;
+    private View mColorView;
+
+    public ColorPreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+
+        TypedArray a = context.getTheme().obtainStyledAttributes(attrs, R.styleable
+                .ColorPreference, 0, 0);
+        try {
+            int id = a.getResourceId(R.styleable.ColorPreference_colors, 0);
```

Comment:
Add a reasonable default array of colors, or throw an exception if `colors` is not present